zp-relayer/workers/sentTxWorker.ts

```diff
@@ -256,7 +250,7 @@ export async function createSentTxWorker<T extends EstimationType>(gasPrice: Gas
         jobLogger.error('Too many unsuccessful re-sends')
       }
 
-      await handleResend(txConfig, gasPrice, job, jobLogger)
+      await handleResend(txConfig, txManager, job, jobLogger)
```

Comment:
Currently whole batch will be considered as invalid and "thrown away".

Reply:
It seems that we need to re-queue some or all direct deposits back into the redis cache, so that valid deposits are not lost?

Reply:
We need to think of possible revert cases. As tx is prepared by relayer we can assume that SNARK proofs will be valid. Another revert option is when we use deposit with a status other than "Pending", but this check is already done in worker validation stage. For now, it looks like a revert is impossible unless there is a problem with relayer proving functionality. Do you have any other ideas when we can see a revert (and how we should determine bad deposits)?

Reply:
We can easily get a revert if one of the previous transactions was reverted on-chain, because the tree proof will become invalid.
Other possibility is related to refunded deposits, but it will only happen if we have small-enough on-chain timeouts for processing.

I think everything should be fine, if we just push all deposits back into the queue, and then revalidate them within the next batch. wdyt?

Reply:
Yeah, agree with this.

Reply:
Done
Added a description in PR comments below